src/main/java/school/lemon/changerequest/java/container/ContainerImpl.java

```diff
@@ -1,38 +1,65 @@
 package school.lemon.changerequest.java.container;
 
-public class ContainerImpl implements Container{
+public class ContainerImpl implements Container {
+
+    private Integer[] container = new Integer[INITIAL_ARRAY_SIZE];
+    private int containerSize = 0;
 
     @Override
     public int size() {
-        // TODO: please implement me
-        return 0;
+        return containerSize;
     }
 
     @Override
     public void clear() {
-        // TODO: please implement me
+        containerSize = 0;
     }
 
     @Override
     public Integer get(int index) {
-        // TODO: please implement me
-        return null;
+        if (index < 0 || index >= containerSize)
+            return null;
+        else
+            return container[index];
     }
 
     @Override
     public void add(int element) {
-        // TODO: please implement me
+        if (containerSize == container.length) {
+            Integer[] tmp = new Integer[containerSize * 2];
+            System.arraycopy(container, 0, tmp, 0, containerSize);
+            container = tmp;
+        }
+        container[containerSize] = element;
+        containerSize++;
     }
 
     @Override
     public boolean add(int element, int index) {
-        // TODO: please implement me
-        return false;
+        if (index < 0 || index > containerSize)
+            return false;
+        if (containerSize == container.length) {
+            Integer[] tmp = new Integer[containerSize * 2];
+            System.arraycopy(container, 0, tmp, 0, index);
+            tmp[index] = element;
+            System.arraycopy(container, index, tmp, index + 1, containerSize - index);
+            container = tmp;
+        } else {
+            for (int i = containerSize; i > index; i--)
+                container[i] = container[i - 1];
+            container[index] = element;
+        }
+        containerSize++;
+        return true;
+
     }
 
     @Override
     public boolean remove(int index) {
-        // TODO: please implement me
-        return false;
+        if (index < 0 || index >= containerSize)
```

Comment:
This condition in the `if` clause is duplicated 3 times — you have to extract it to a separate method and re-use it! What if you'd have 200 additional methods and the condition would change?